routingpy/valhalla_attributes.py

```diff
@@ -0,0 +1,421 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2021 GIS OPS UG
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+"""
+:class:`Expansion` returns expansion results.
+"""
+from enum import Enum
+from typing import List, Union, Tuple, Optional
+
+from routingpy.utils import decode_polyline6
+
+
+class MatchDiscontinuity(str, Enum):
+    BEGIN = "begin"
+    END = "end"
+    NONE = ""
+
+
+class MatchType(str, Enum):
+    UNMATCHED = "unmatched"
+    MATCHED = "matched"
+    INTERPOLATED = "interpolated"
+    NONE = ""
+
+
+class Traversability(str, Enum):
+    FORWARD = "forward"
+    BACKWARD = "backward"
+    BOTH = "both"
+    NONE = ""
+
+
+class DrivingSide(str, Enum):
+    RIGHT = "right"
+    LEFT = "left"
+    NONE = ""
+
+
+class Sidewalk(str, Enum):
+    RIGHT = "right"
+    LEFT = "left"
+    BOTH = "both"
+    NONE = ""
+
+
+class Surface(str, Enum):
+    PAVED_SMOOTH = "paved_smooth"
+    PAVED = "paved"
+    PAVED_ROUGH = "paved_rough"
+    COMPACTED = "compacted"
+    DIRT = "dirt"
+    GRAVEL = "gravel"
+    PATH = "path"
+    IMPASSABLE = "impassable"
+    NONE = ""
+
+
+class RoadClass(str, Enum):
+    MOTORWAY = "motorway"
+    TRUCK = "trunk"
+    PRIMARY = "primary"
+    SECONDARY = "secondary"
+    TERTIARY = "tertiary"
+    UNCLASSIFIED = "unclassified"
+    RESIDENTIAL = "residential"
+    SERVICE_OTHER = "service_other"
+    NONE = ""
+
+
+class Use(str, Enum):
+    TRAM = "tram"
+    ROAD = "road"
+    RAMP = "ramp"
+    TURN_CHANNEL = "turn_channel"
+    TRACK = "track"
+    DRIVEWAY = "driveway"
+    ALLEY = "alley"
+    PARKING_AISLE = "parking_aisle"
+    EMERGENCY_ACCESS = "emergency_access"
+    DRIVE_TROUGH = "drive_through"
+    CULDESAC = "culdesac"
+    CYCLEWAY = "cycleway"
+    MOUNTAIN_BIKE = "mountain_bike"
+    SIDEWALK = "sidewalk"
+    FOOTWAY = "footway"
+    STEPS = "steps"
+    OTHER = "other"
+    RAIL_FERRY = "rail-ferry"
+    FERRY = "ferry"
+    RAIL = "rail"
+    BUS = "bus"
+    RAIL_CONNECTION = "rail_connection"
+    BUS_CONNECTION = "bus_connnection"
+    TRANSIT_CONNECTION = "transit_connection"
+    NONE = ""
+
+
+class MatchedEdge:
```

Comment:
this is a single edge object with most interesting properties set. could be extended in future, but good enough for NA